src/main/java/org/folio/reader/JPathSyntaxEntityReader.java

```diff
@@ -45,6 +46,59 @@ public JPathSyntaxEntityReader(JsonObject entity) {
     );
   }
 
+  @Override
+  protected RuleValue readSimpleValue(Rule rule, DataSource dataSource) {
+    populateMetadata(rule);
+    List<ValueWrapper> valueWrappers = readMatrix(rule).get(0).getValue();
+    if (valueWrappers.isEmpty()) {
+      return MissingValue.getInstance();
+    } else {
+      Optional<ValueWrapper> optionalNonNullValue = valueWrappers.stream().filter(valueWrapper -> valueWrapper.getValue() != null).findFirst();
+      if (optionalNonNullValue.isPresent()) {
+        SimpleValue simpleValue = buildSimpleValue(dataSource, valueWrappers, optionalNonNullValue.get().getValue());
+        return simpleValue;
+      } else {
+        return MissingValue.getInstance();
+      }
+    }
+  }
+
+  /**
+   * Builds a SimpleValue, could be StringValue or ListValue
+   */
+  private SimpleValue buildSimpleValue(DataSource dataSource, List<ValueWrapper> valueWrappers, Object nonNullValue) {
+    boolean isStringRuleValue = false;
+    boolean isListRuleValue = false;
```

Comment:
I don't think that usage of these two fields is necessary,  you can just use the condition bellow in the if/else statement